src/components/views/rooms/RoomHeader.tsx

```diff
@@ -48,22 +51,287 @@ import { BetaPill } from "../beta/BetaCard";
+
+    let menu: JSX.Element | null = null;
+    if (menuOpen) {
+        const buttonRect = buttonRef.current!.getBoundingClientRect();
+        menu = <IconizedContextMenu {...aboveLeftOf(buttonRect)} onFinished={closeMenu}>
+            <IconizedContextMenuOptionList>
+                <IconizedContextMenuOption label={_t("Video call (Jitsi)")} onClick={onJitsiClick} />
+                <IconizedContextMenuOption label={_t("Video call (Element Call)")} onClick={onElementClick} />
+            </IconizedContextMenuOptionList>
+        </IconizedContextMenu>;
+    }
+
+    return <>
+        <AccessibleTooltipButton
+            inputRef={buttonRef}
+            className="mx_RoomHeader_button mx_RoomHeader_videoCallButton"
+            onClick={onClick}
+            title={_t("Video call")}
+            tooltip={tooltip ?? _t("Video call")}
+            disabled={disabled || busy}
+        />
+        { menu }
+    </>;
+};
+
+interface CallButtonsProps {
+    room: Room;
+}
+
+// The header buttons for placing calls have become stupidly complex, so here
+// they are as a separate component
+const CallButtons: FC<CallButtonsProps> = ({ room }) => {
+    const [busy, setBusy] = useState(false);
+    const showButtons = useSettingValue<boolean>("showCallButtonsInComposer");
+    const groupCallsEnabled = useFeatureEnabled("feature_group_calls");
+    const videoRoomsEnabled = useFeatureEnabled("feature_video_rooms");
+    const isVideoRoom = useMemo(() => videoRoomsEnabled && calcIsVideoRoom(room), [videoRoomsEnabled, room]);
+    const useElementCallExclusively = useMemo(() => SdkConfig.get("element_call").use_exclusively, []);
+
+    const hasLegacyCall = useEventEmitterState(
+        LegacyCallHandler.instance,
+        LegacyCallHandlerEvent.CallsChanged,
+        useCallback(() => LegacyCallHandler.instance.getCallForRoom(room.roomId) !== null, [room]),
+    );
+
+    const widgets = useWidgets(room);
+    const hasJitsiWidget = useMemo(() => widgets.some(widget => WidgetType.JITSI.matches(widget.type)), [widgets]);
+
+    const hasElementCall = useCall(room.roomId) !== null;
+
+    const [functionalMembers, mayEditWidgets, mayCreateElementCalls] = useTypedEventEmitterState(
+        room,
+        RoomStateEvent.Update,
+        useCallback(() => [
+            getJoinedNonFunctionalMembers(room),
+            room.currentState.mayClientSendStateEvent("im.vector.modular.widgets", room.client),
+            room.currentState.mayClientSendStateEvent(ElementCall.CALL_EVENT_TYPE.name, room.client),
+        ], [room]),
+    );
+
+    const makeVoiceCallButton = (behavior: VoiceCallButtonProps["behavior"]): JSX.Element =>
+        <VoiceCallButton room={room} busy={busy} setBusy={setBusy} behavior={behavior} />;
+    const makeVideoCallButton = (behavior: VideoCallButtonProps["behavior"]): JSX.Element =>
+        <VideoCallButton room={room} busy={busy} setBusy={setBusy} behavior={behavior} />;
+
+    if (isVideoRoom || !showButtons) {
```

Comment:
```suggestion
    if (!showButtons) {
```

Why not ^?

Reply:
The video rooms designs have always called for the call buttons to be hidden from the room header, and I don't want to change that here